pkgs/applications/misc/cura/lulzbot.nix

```diff
@@ -1,58 +1,42 @@
-{ stdenv, fetchurl, dpkg, bash, python27Packages }:
+{ lib, fetchgit, curaengineLulzbot, cmake, python3Packages, qtbase, qtquickcontrols2 }:
 
-let
-  py = python27Packages;
-in
-stdenv.mkDerivation rec {
+python3Packages.buildPythonApplication rec {
   name = "cura-lulzbot-${version}";
-  version = "15.02.1-1.03-5064";
+  version = "3.6.8";
 
-  src =
-    if stdenv.hostPlatform.system == "x86_64-linux" then
-      fetchurl {
-        url = "https://download.alephobjects.com/ao/aodeb/dists/jessie/main/binary-amd64/cura_${version}_amd64.deb";
-        sha256 = "1gsfidg3gim5pjbl82vkh0cw4ya253m4p7nirm8nr6yjrsirkzxg";
-      }
-    else if stdenv.hostPlatform.system == "i686-linux" then
-      fetchurl {
-        url = "http://download.alephobjects.com/ao/aodeb/dists/jessie/main/binary-i386/cura_${version}_i386.deb";
-        sha256 = "0xd3df6bxq4rijgvsqvps454jkc1nzhxbdzzj6j2w317ppsbhyc1";
-      }
-    else throw "${name} is not supported on ${stdenv.hostPlatform.system}";
-
-  python_deps = with py; [ pyopengl pyserial numpy wxPython30 power setuptools ];
-  pythonPath = python_deps;
-  propagatedBuildInputs = python_deps;
-  buildInputs = [ dpkg bash py.wrapPython ];
-
-  phases = [ "unpackPhase" "installPhase" ];
-  unpackPhase = "dpkg-deb -x ${src} ./";
-
-  installPhase = ''
-    mkdir -p $out/bin
-    cp -r usr/share $out/share
-    find $out/share -type f -exec sed -i 's|/usr/share/cura|$out/share/cura|g' "{}" \;
-
-    cat <<EOT > $out/bin/cura
-    #!${bash}/bin/bash
-    PYTHONPATH=$PYTHONPATH:$out/share/cura ${py.python}/bin/python $out/share/cura/cura.py "\$@"
-    EOT
+  src = fetchgit {
+    url = https://code.alephobjects.com/source/cura-lulzbot.git;
+    rev = "0902c91c8c340adbf645465076a63563ec689de4";
+    sha256 = "16pzifwpki06mkkj5ymkxjvcdppm2hv3cr2fpa5f770cdxnxbzzl";
+  };
 
-    chmod 555 $out/bin/cura
+  format = "other"; # using cmake to build
+  buildInputs = [ qtbase qtquickcontrols2 ];
+  propagatedBuildInputs = with python3Packages; [
+    libsavitarLulzbot numpy-stl pyserial requests uraniumLulzbot zeroconf
+  ];
+  nativeBuildInputs = [ cmake python3Packages.wrapPython ];
+
+  cmakeFlags = [
+    "-DURANIUM_DIR=${python3Packages.uranium.src}"
+    "-DCURA_VERSION=${version}"
+  ];
+
+  postPatch = ''
+    sed -i 's,/python''${PYTHON_VERSION_MAJOR}/dist-packages,/python''${PYTHON_VERSION_MAJOR}.''${PYTHON_VERSION_MINOR}/site-packages,g' CMakeLists.txt
+    sed -i 's, executable_name = .*, executable_name = "${curaengineLulzbot}/bin/CuraEngine",' plugins/CuraEngineBackend/CuraEngineBackend.py
   '';
 
-  meta = with stdenv.lib; {
-    description = "3D printing host software for the Lulzbot";
-
-     longDescription = ''
-       Cura LulzBot Edition is a fork of the 3D printing/slicing
-       software from Ultimaker, with changes to support 3D printers
-       from Aleph Objects.
-     '';
+  preFixup = ''
+    sed -i 's,^import cura.CuraApplication$,import Savitar; import cura.CuraApplication,' "$out/bin/cura-lulzbot"
```

Comment:
`substituteInPlace` is preferred here because it complains if no substitution has been made (at least as a logging message).

Reply:
Thank you for the review. I've squashed the change 80b975d75c0 in, and checked that the output file is identical / the executable still runs correctly.